tests/logitech_receiver/test_profiles.py

```diff
@@ -0,0 +1,87 @@
+import re
+
+import logitech_receiver.hidpp20 as hidpp20
+import yaml
+
+
+def test_led_setting_bytes():
+    ebytes = bytes.fromhex("0A01020300500407000000")
+    setting = hidpp20.LEDEffectSetting.from_bytes(ebytes)
+    assert setting.ID == 0x0A, "LED setting ID read incorrectly"
+    assert setting.color == 0x010203, "LED setting color read incorrectly"
+    assert setting.period == 0x0050, "LED setting color read incorrectly"
+    assert setting.form == 0x04, "LED setting color read incorrectly"
+    assert setting.intensity == 0x07, "LED setting color read incorrectly"
+    bytes_out = setting.to_bytes()
+    assert ebytes == bytes_out, "LED setting bytes out not same as bytes in"
+
+
+def test_led_setting_yaml():
+    ebytes = bytes.fromhex("0A01020300500407000000")
+    eyaml = (
+        "!LEDEffectSetting {ID: !NamedInt {name: Breathe, value: 0xa}, color: 0x10203, "
+        "form: 0x4, intensity: 0x7, period: 0x50} "
+    )
+    setting = hidpp20.LEDEffectSetting.from_bytes(ebytes)
+    assert setting.ID == 0x0A, "LED setting ID read incorrectly"
+    assert setting.color == 0x010203, "LED setting color read incorrectly"
+    assert setting.period == 0x0050, "LED setting color read incorrectly"
+    assert setting.form == 0x04, "LED setting color read incorrectly"
+    assert setting.intensity == 0x07, "LED setting color read incorrectly"
+    yaml_out = yaml.dump(setting)
+    assert eyaml == re.compile(r"\s+").sub(" ", yaml_out)
+    setting = yaml.safe_load(eyaml)
+    assert setting.to_bytes() == ebytes
+
+
+def test_button_bytes():
+    bbytes = bytes.fromhex("8000FFFF")
+    button = hidpp20.Button.from_bytes(bbytes)
+    assert button.behavior == 0x8, "button profile behaviour read incorrectly"
+    assert button.type == 0x00, "button profile type read incorrectly"
+    bytes_out = button.to_bytes()
+    assert bbytes == bytes_out, "button bytes out not same as bytes in"
+
+    bbytes = bytes.fromhex("900aFF00")
+    button = hidpp20.Button.from_bytes(bbytes)
+    assert button.behavior == 0x9, "button profile behaviour read incorrectly"
+    bytes_out = button.to_bytes()
+    assert bbytes == bytes_out, "button bytes out not same as bytes in"
+
+    bbytes = bytes.fromhex("80020454")
+    button = hidpp20.Button.from_bytes(bbytes)
+    assert button.behavior == 0x8, "button profile behaviour read incorrectly"
+    assert button.modifiers == 0x04, "button profile type read incorrectly"
+    bytes_out = button.to_bytes()
+    assert bbytes == bytes_out, "button bytes out not same as bytes in"
+
+
+pbytes = bytes.fromhex(
```

Comment:
I slightly favour a function here, that at least ensures, that each test starts with unmodified data.

def get_profile_bytes()